ci-operator/step-registry/upi/conf/gcp/xpn/upi-conf-gcp-xpn-commands.sh

```diff
@@ -24,15 +24,44 @@ set -o nounset
 set -o errexit
 set -o pipefail
 
-HOST_PROJECT="openshift-dev-installer"
-CLUSTER_NETWORK="https://www.googleapis.com/compute/v1/projects/openshift-dev-installer/global/networks/installer-shared-vpc"
-CONTROL_SUBNET="https://www.googleapis.com/compute/v1/projects/openshift-dev-installer/regions/us-east1/subnetworks/installer-shared-vpc-subnet-1"
-COMPUTE_SUBNET="https://www.googleapis.com/compute/v1/projects/openshift-dev-installer/regions/us-east1/subnetworks/installer-shared-vpc-subnet-2"
-COMPUTE_SERVICE_ACCOUNT="do-not-delete-ci-xpn@openshift-gce-devel-ci.iam.gserviceaccount.com"
-CONTROL_SERVICE_ACCOUNT="do-not-delete-ci-xpn@openshift-gce-devel-ci.iam.gserviceaccount.com"
-PRIVATE_ZONE_NAME="ci-op-xpn-private-zone"
+if [[ -f "${CLUSTER_PROFILE_DIR}/ci-xpn.json" ]]; then
 
-cat > "${SHARED_DIR}/xpn.json" << EOF
+    echo "Using XPN configurations of the cluster profile 'gcp-qe'..."
+    HOST_PROJECT="openshift-qe-shared-vpc"
+    CLUSTER_NETWORK="https://www.googleapis.com/compute/v1/projects/openshift-qe-shared-vpc/global/networks/installer-shared-vpc"
+    CONTROL_SUBNET="https://www.googleapis.com/compute/v1/projects/openshift-qe-shared-vpc/regions/us-central1/subnetworks/installer-shared-vpc-subnet-1"
+    COMPUTE_SUBNET="https://www.googleapis.com/compute/v1/projects/openshift-qe-shared-vpc/regions/us-central1/subnetworks/installer-shared-vpc-subnet-2"
+    COMPUTE_SERVICE_ACCOUNT="do-not-delete-ci-xpn@openshift-qe.iam.gserviceaccount.com"
+    CONTROL_SERVICE_ACCOUNT="do-not-delete-ci-xpn@openshift-qe.iam.gserviceaccount.com"
+    PRIVATE_ZONE_NAME="ci-op-xpn-private-zone"
+    BASE_DOMAIN="qe-shared-vpc.qe.gcp.devcluster.openshift.com"
```

Comment:
`qe.gcp.devcluster.openshift.com` (`BASE_DOMAIN`) is retrieved from `public_hosted_zone` or `BASE_DOMAIN` system env like @jianlinliu mentioned, but we write hard code here, is there a better solution?  put all project-specified resources (`HOST_PROJECT`, `COMPUTE_SERVICE_ACCOUNT`, `PRIVATE_ZONE_NAME`, etc. ) in vault or system env?

Reply:
I didn't put the XPN settings in vault because existing dev's profile doesn't do it that way.

Reply:
oh, I just saw that.

Reply:
If we can save these project-specified resources in vault, that would be perfect, e.g: define `xpn_host_project`, `xpn_compute_service_account` in vault, then retrieve them, determinate to override or not. Something like:
```
HOST_PROJECT="openshift-dev-installer"
if [ -f "${CLUSTER_PROFILE_DIR}/xpn_host_project" ]; then
  HOST_PROJECT=$(< "${CLUSTER_PROFILE_DIR}/xpn_host_project")
fi
CLUSTER_NETWORK="https://www.googleapis.com/compute/v1/projects/openshift-dev-installer/global/networks/installer-shared-vpc"
if [ -f "${CLUSTER_PROFILE_DIR}/xpn_cluster_network" ]; then
  CLUSTER_NETWORK=$(<"${CLUSTER_PROFILE_DIR}/xpn_cluster_network")
fi
<--snip-->
cat > "${SHARED_DIR}/xpn.json" << EOF
{
    "hostProject": "${HOST_PROJECT}",
    "clusterNetwork": "${CLUSTER_NETWORK}",
    "computeSubnet": "${COMPUTE_SUBNET}",
    "controlSubnet": "${CONTROL_SUBNET}",
    "computeServiceAccount": "${COMPUTE_SERVICE_ACCOUNT}",
    "controlServiceAccount": "${CONTROL_SERVICE_ACCOUNT}",
    "privateZoneName": "${PRIVATE_ZONE_NAME}"
}
```

Reply:
Once that, I guess we can drop several `[[ -f "${CLUSTER_PROFILE_DIR}/qe-ci-xpn.json"` condition.

Reply:
I'd put all XPN settings (except credentials) in "${CLUSTER_PROFILE_DIR}/qe_gce_xpn.json".

Reply:
hmm, `qe_gce_xpn.json` is used to sa auth login, right?

Reply:
`qe_gce_xpn_creds.json` is used for sa login.

> I'd put all XPN settings (except credentials) in "${CLUSTER_PROFILE_DIR}/qe_gce_xpn.json".

agree with this, this allows us to change settings without updating code,  and easy to management.

>  define xpn_host_project, xpn_compute_service_account in vault, then retrieve them, determinate to override or not. Something like:

or get all infos from on json file
```
if [ -f "${CLUSTER_PROFILE_DIR}/qe_gce_xpn.json" ]; then
  HOST_PROJECT=$(jq -r '. hostProject' "${CLUSTER_PROFILE_DIR}/qe_gce_xpn.json")
  CLUSTER_NETWORK=$(jq -r '. clusterNetwork' "${CLUSTER_PROFILE_DIR}/qe_gce_xpn.json")
fi```